ci/__main__.py

```diff
@@ -0,0 +1,686 @@
+#     "*/tests",
+# }
+
+environment_vars = {
+    # 'pyside2': {'ETS_TOOLKIT': 'qt', 'QT_API': 'pyside2'},
+    # 'pyside6': {'ETS_TOOLKIT': 'qt', 'QT_API': 'pyside6'},
+    'pyqt5': {"ETS_TOOLKIT": "qt", "QT_API": "pyqt5"},
+    # 'pyqt6': {"ETS_TOOLKIT": "qt", "QT_API": "pyqt6"},
+    # 'wx': {'ETS_TOOLKIT': 'wx'},
+    # 'null': {'ETS_TOOLKIT': 'null'},
+}
+
+# Location of documentation files
+HERE = os.path.dirname(__file__)
+DOCS_DIR = os.path.join(HERE, "docs")
+
+# Location of news fragment for creating changelog.
+NEWS_FRAGMENT_DIR = os.path.join(DOCS_DIR, "releases", "upcoming")
+
+# Location of the Changelog file.
+CHANGELOG_PATH = os.path.join(HERE, "CHANGES.txt")
+
+
+def normalize(name):
+    return name.replace("_", "-")
+
+
+# Ensure that "-h" is supported for getting help.
+CONTEXT_SETTINGS = dict(
+    help_option_names=["-h", "--help"],
+    token_normalize_func=normalize,
+)
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--runtime', default=DEFAULT_RUNTIME)
+@click.option('--toolkit', default=DEFAULT_TOOLKIT)
+@click.option('--environment', default=None)
+@click.option(
+    "--editable/--not-editable",
+    default=False,
+    help="Install main package in 'editable' mode?  [default: --not-editable]",
+)
+@click.option('--source/--no-source', default=False)
+def install(runtime, toolkit, environment, editable, source):
+    """Install project and dependencies into a clean EDM environment."""
+    parameters = get_parameters(runtime, toolkit, environment)
+    packages = (dependencies | ci_dependencies)
+
+    # edm commands to setup the development environment
+    commands = [
+        "edm environments create {environment} --force --version={runtime}"
+    ]
+
+    commands.extend(
+        [
+            "edm install -y -e {environment} " + " ".join(packages),
+            # "edm run -e {environment} -- pip install --force-reinstall -r
+            # ci-src-requirements.txt --no-dependencies",
+            "edm run -e {environment} -- python setup.py clean --all",
+        ]
+    )
+
+    # pip install pyqt5, pyqt6, pyside2 and pyside6, because we don't have them
+    # in EDM yet
+    if toolkit == 'pyside2':
+        commands.append("edm run -e {environment} -- pip install pyside2")
+    elif toolkit == 'pyqt6':
+        commands.append("edm run -e {environment} -- pip install pyqt6")
+    elif toolkit == 'pyside6':
+        # On Linux and macOS, some versions of PySide6 between 6.2.2 and 6.3.0
+        # are unimportable on Python 3.6 and Python 3.7. See
+        # https://bugreports.qt.io/browse/PYSIDE-1797. It may be possible to
+        # remove this workaround once PySide6 6.3.1 or later is released.
+        if sys.platform in {'darwin', 'linux'}:
+            commands.append(
+                "edm run -e {environment} -- pip install pyside6<6.2.2")
+        else:
+            commands.append("edm run -e {environment} -- pip install pyside6")
```

Comment:
I'm still not sure about why we would need this logic. If we expect tutorial attendees to use the CI module, then allowing them to specify the toolkit can be a useful escape hatch if pyqt5 doesn't work for them.

Reply:
Hey @siddhantwahal , I'm not sure I understand your meaning here...